ext/openssl/ossl_x509store.c

```diff
@@ -130,7 +130,7 @@ static const rb_data_type_t ossl_x509store_type = {
     {
         ossl_x509store_mark, ossl_x509store_free,
     },
-    0, 0, RUBY_TYPED_FREE_IMMEDIATELY,
+    0, 0, RUBY_TYPED_FREE_IMMEDIATELY | RUBY_TYPED_WB_PROTECTED,
```

Comment:
The `ossl_x509store_alloc` is calling the `NewX509Store` calling the `ossl_x509store_type`. Below is soon after calling the `NewX509Store`. Is there a way to check the internal expression of the `obj` to confirm how the code above changed it?

```
$ cat test.rb 
require "openssl"
p OpenSSL::X509::Store.new
```

```
$ gdb --args ruby -Ilib test.rb
...
(gdb) f
#0  ossl_x509store_alloc (klass=140737043491920)
    at ../../../../ext/openssl/ossl_x509store.c:174
(gdb) l
169	{
170	    X509_STORE *store;
171	    VALUE obj;
172	
173	    obj = NewX509Store(klass);
174	    if ((store = X509_STORE_new()) == NULL)
175	        ossl_raise(eX509StoreError, "X509_STORE_new");
176	    SetX509Store(obj, store);
177	
178	    return obj;
(gdb) p obj
$6 = 140737043392360
```

Reply:
> Is there a way to check the internal expression of the obj to confirm how the code above changed it?

Yes, you could cast that `VALUE` to `RTypedData` etc.

But it's simpler to do it from the Ruby side:

master:

```ruby
>> require 'openssl'
=> true
>> require 'objspace'
=> true
>> puts ObjectSpace.dump(OpenSSL::X509::Store.new)
{"address":"0x10f265290", "type":"DATA", "class":"0x10f22c530", "struct":"OpenSSL/X509/STORE", "memsize":88}
```

this branch:

```
>> puts ObjectSpace.dump(OpenSSL::X509::Store.new)
{"address":"0x10450c030", "type":"DATA", "class":"0x1045c77b8", "struct":"OpenSSL/X509/STORE", "memsize":88, "flags":{"wb_protected":true}}
```

Note the `wb_protected` flag.

Reply:
Okay. Thanks for the info!